nikola/nikola.py

```diff
@@ -2374,11 +2376,16 @@ def generic_index_renderer(self, lang, posts, indexes_title, template_name, cont
                 lists.append(posts[:kw["index_display_post_count"]])
                 posts = posts[kw["index_display_post_count"]:]
         num_pages = len(lists)
+        displayed_page_numbers = [utils.get_displayed_page_number(i, num_pages, self) for i in range(max(num_pages, 1))]
+        page_links = [page_link(i, displayed_page_numbers[i], num_pages, False) for i in range(max(num_pages, 1))]
+        if kw['show_index_page_navigation']:
+            temp_map = {page_number - 1: link for page_number, link in zip(displayed_page_numbers, page_links)}
```

Comment:
The list is not sorted already: in case `INDEXES_STATIC == True`, `displayed_page_numbers` will be `[num_pages, 1, 2, 3, ..., num_pages - 1]`. In case `INDEXES_STATIC == False`, it will be `[1, 2, ..., num_pages]` (see `utils.get_displayed_page_number`). To avoid having to duplicate that kind of strange logic in here, I'm using the map.

In case `num_pages == 0`, `page_links_context` should have length 0 (and not length 1 as `displayed_page_numbers` and `page_links`). This isn't the case for `[i[1] for i in sorted(temp_map.values())]`, which will have length 1.

Anyway, in case `num_pages == 0`, the whole thing is not working that well anyway. In case `INDEXES_PRETTY_PAGE_URL` is `True`, a redirect to a non-existing page will be created, and in any case, the main index page is not created. How about fixing that here as well?